zaza/openstack/charm_tests/ceph/tests.py

```diff
@@ -413,6 +413,44 @@ def test_blocked_when_non_pristine_disk_appears(self):
 
         zaza_model.wait_for_application_states()
 
+    def test_active_after_pristine_block(self):
+        """Test after a non-pristine disk was configured and deconfigured.
+
+        This is to test the case of LP #1988088 where a charm would not come
+        out of blocked state after a non-pristine disk was deconfigured
+
+        We are using the same ephemeral disk as in the previous test, see
+        above `test_blocked_when_non_pristine_disk_appears` for details.
+        """
+        logging.info('Ensuring default configuration...')
+        juju_service = 'ceph-osd'
+        set_default = {
+            'ephemeral-unmount': '',
```

Comment:
It feels like this set_default should look at whatever's currently set in the model rather than just using charm defaults

Reply:
I've now added two calls to store app config away and re-store it at the end of the test method so we're going back to where we started -- does this address your concerns?